velox/row/UnsafeRowDeserializers.h

```diff
@@ -208,8 +208,9 @@ struct StructBatchIterator : UnsafeRowDataBatchIterator {
    */
   const std::vector<std::optional<std::string_view>>& nextColumnBatch() {
     const TypePtr& type = childTypes_[idx_];
-    std::size_t fixedSize =
-        type->isFixedWidth() ? serializedSizeInBytes(type) : 0;
+    std::size_t fixedSize = type->isFixedWidth() && !type->isLongDecimal()
```

Comment:
This is Spark UnsafeRow deserializer https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java#L369, it uses variable-width binary to represent long decimal.
And this is serialize decimal(BigInteger in java) https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java#L276

Reply:
I found this description of BigInteger serialization. It would be nice to include it in the doc and explain that it is actually variable-width.

https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html#toByteArray--

```
Returns a byte array containing the two's-complement representation of this BigInteger. The byte array will be in big-endian byte-order: the most significant byte is in the zeroth element. The array will contain the minimum number of bytes required to represent this BigInteger, including at least one sign bit, which is (ceil((this.bitLength() + 1)/8)).
```